packages/rum/src/domain/record/record.ts

```diff
@@ -86,19 +88,31 @@ export function record(options: RecordOptions): RecordAPI {
   takeFullSnapshot()
 
   const stopObservers = initObservers({
+    lifeCycle: options.lifeCycle,
     mutationController,
     defaultPrivacyLevel: options.defaultPrivacyLevel,
     inputCb: (v) => emit(assembleIncrementalSnapshot<InputData>(IncrementalSource.Input, v)),
     mediaInteractionCb: (p) =>
       emit(assembleIncrementalSnapshot<MediaInteractionData>(IncrementalSource.MediaInteraction, p)),
     mouseInteractionCb: (d) =>
-      emit(assembleIncrementalSnapshot<MouseInteractionData>(IncrementalSource.MouseInteraction, d)),
+      emit(
+        assign(
+          { recordId: d.recordId },
+          assembleIncrementalSnapshot<MouseInteractionData>(IncrementalSource.MouseInteraction, d)
+        )
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏It looks like we derive a bit from the pattern here. I think having such callbacks is impractical and we might want to simplify this in the future. For now, focusing on `mouseInteractionCb` and `frustrationCb`: could you change those callbacks so they get respectively a `IncrementalSnapshotRecord` and `FrustrationRecord` as argument, so we don't need an intermediary data model anymore?